### PR TITLE
Latest codes

### PR DESCRIPTION
**Pull request by joshuaobinna**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/node_modules
 package-lock.json
-node_modules
-/*Agro_easy
+/New folder
+.gitconfig
```
Comment (Way2nnadi) [inline .gitignore]: remove `/New Folder` and `.gitconfig` from the `. gitignore` they have nothing to do with this repo

File: README.md [hunk 2]
```diff
@@ -1 +0,0 @@
-# Agroeasy
```
Comment (Way2nnadi) [inline README.md]: Why did you delete the README.md

File: src/App.js [hunk 3]
```diff
@@ -3,17 +3,18 @@ import { BrowserRouter as Router, Route } from 'react-router-dom';
...
 
-/*
-The principal component for housing sub-components (like home, about and
-other pages) for easy navigation
-*/
+
+
+//The principal component for housing sub-components (like home, about and
+//other pages) for easy navigation
+
```
Comment (Way2nnadi) [inline src/App.js]: The multiline commenting with `/***/` is correct, change it back to that

File: src/components/anotherlist.js [hunk 4]
```diff
@@ -1,198 +1,78 @@
...
-                   </Row>
-                         <div className="link-to-more">
-                               <Link to='/more-products/'>
-                                    More
-                               </Link>
-                         </div>
-              </Jumbotron>
-        </Container>
+        <div>
+           <Jumbotron className="show-products-available" style={{backgroundColor: 'ash'}}>
```
Comment (Way2nnadi) [inline src/components/anotherlist.js]: I would wondering, why is this whole container inside of a `Jumbotron`?

File: src/components/anotherlist.js [hunk 5]
```diff
@@ -1,198 +1,78 @@
...
+                         <Row2/>
+                       </Row>
+                       <br />
+                       <Row>
+                         <Row3 />
+                       </Row>
+                       <br />
+                       <Row>
+                         <Row4 />
+                       </Row>
```
Comment (Way2nnadi) [inline src/components/anotherlist.js]: Your `Row1/2/3/4` should already be contained within a `Row` component no need repeat yourself here multiple times. Your `line 45 - 60` should look something like 
```
<Row1 />
<Row2 />
<Row3/>
<Row4 />
```

Comment (Way2nnadi) [reply src/components/anotherlist.js]: also remove the `<br />` use css to do the spacing

File: src/components/constant.js [hunk 6]
```diff
@@ -32,3 +32,131 @@ export const Ginger = "/ginger/";
...
+        email: "myemail@example.com"
+    },
+    {
+        name: "Ripe banana",
+        cost: "#500",
+        quantity: "200 bunches",
+        location: "Ore, ondo state",
+        email: "myemail@example.com"
+    }
+]
```
Comment (Way2nnadi) [inline src/components/constant.js]: Very good, this is a temporary evil until we have the database hooked up properly

File: src/components/contact-us.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                  <ModalFooter>
+                     <Button color="danger" onClick={this.toggle} email="exampleEmail.com">Submit</Button>{' '}
+                     <Button color="secondary" onClick={this.toggle}>Cancel</Button>
+                  </ModalFooter>
+              </Modal>
+         </div>
+    );
+  }
+}
+export default ContactUs;
```
Comment (Way2nnadi) [inline src/components/contact-us.js]: This was not a part of this pull request, but its good job on implementing it. You want to keep scope of a pull request limited to feature or bug that you are working on

File: src/components/grouped-items/fruitandvegetables.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+                                   <h6>Fruits / vegetables</h6>
+                                   <li><Link to={Cabbage} style={{color: 'DarkGray'}}>Cabbage</Link></li>
+                                   <li><Link to={Lettuse} style={{color: 'DarkGray'}}>Lettuse</Link></li>
+                                   <li><Link to={Tomatoes} style={{color: 'DarkGray'}}>Tomatoes</Link></li>
+                                   <li><Link to={Oranges} style={{color: 'DarkGray'}}>Oranges</Link></li>
+                                   <li><Link to={Plantain} style={{color: 'DarkGray'}}>Plaintain</Link></li>
+                                 </ul>
+                          </Col>
+                     </Row>
+                 </Container>
```
Comment (Way2nnadi) [inline src/components/grouped-items/fruitandvegetables.js]: This component looks the same as the next two following components below. You should encapsulate this logic into one component.

File: src/components/grouped-items/livestockpoultry.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+                              <li><Link to={Eggs} style={{color: 'DarkGray'}}>Eggs</Link></li>
+                              <li><Link to={Cattle} style={{color: 'DarkGray'}}>Cattles</Link></li>
+                              <li><Link to={Chicken} style={{color: 'DarkGray'}}>Chicken</Link></li>
+                              <li><Link to={Ram} style={{color: 'DarkGray'}}>Rams</Link></li>
+                           </ul>              
+                        </Col>
+                     </Row>
+              </Container>
+			);
+	}
```
Comment (Way2nnadi) [inline src/components/grouped-items/livestockpoultry.js]: You should encapsulate this logic into one component given that it matches the one above.

File: src/components/grouped-items/spices.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+                           <li><Link to={Garlic} style={{color: 'DarkGray'}}>Garlic</Link></li>
+                           <li><Link to={Pepper} style={{color: 'DarkGray'}}>Pepper</Link></li>
+                           <li><Link to={Curry} style={{color: 'DarkGray'}}>Curry</Link></li>
+                        </ul> 
+                      </Col>
+                   </Row>
+                </Container>
+        );
+    }
+}
```
Comment (Way2nnadi) [inline src/components/grouped-items/spices.js]: You should encapsulate this logic into one component given that it matches the ones above.

File: src/components/rows/product.js [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+                          </ButtonGroup>
+                      </ButtonToolbar>
+                   </p>
+              </div>
+      </div>
+    );
+  }
+}
+
+export default Products;
```
Comment (Way2nnadi) [inline src/components/rows/product.js]: So this was what I was talking about in the meeting. You are giving each item its own modal. Imagine now that you had 100, 1000, 10000 items on the screen and each one has a modal attached to it. The browser would have trouble attaching a modal to each item `div`.

Comment (Way2nnadi) [reply src/components/rows/product.js]: The solution here is to only have one modal and pass it in the appropriate information when you click on an item. Again one modal serving 0 to any number of items on our view
```
Step 1 - move the modal logic to its own component
2 - add modal the newly created modal component to the `anotherlist` component. it should be a sibling component to the `ListItems`
3 - writing a function that you pass into each modal component, that when clicked updates `anotherlist` internal state
4. the state update should tell the modal whether it should be open or not, and what contents it should display

**Comment by Way2nnadi** [comment]
Obinna look at the current file structure of mater, make sure your images are being put in the right place

File: src/components/grouped-items/livestockpoultry.js (hunk 9 above)
Comment (EngrEric) [reply src/components/grouped-items/livestockpoultry.js]: Encapsulation in the sense that one component should house them or wrap them?

**Comment by Way2nnadi** [comment]
You also have a bunch of merge conflicts that you must fix before, I can approve this code to be merged down

**Comment by Way2nnadi** [comment]
Obinna, will be closing this PR for now.

**Comment by joshuaobinna** [comment]
Ok

Sent from Yahoo Mail on Android 
 
  On Mon, Oct 29, 2018 at 17:08, Uchechukwu Nnadi<notifications@github.com> wrote:   
@Way2nnadi commented on this pull request.

In .gitignore:
>  package-lock.json
-node_modules
-/*Agro_easy
+/New folder
+.gitconfig

remove /New Folder and .gitconfig from the . gitignore they have nothing to do with this repo

File: .gitignore (hunk 1 above)
Comment (joshuaobinna) [reply .gitignore]: done

File: src/App.js (hunk 3 above)
Comment (joshuaobinna) [reply src/App.js]: okay

File: src/components/anotherlist.js (hunk 5 above)
Comment (joshuaobinna) [reply src/components/anotherlist.js]: Done

File: src/components/constant.js (hunk 6 above)
Comment (joshuaobinna) [reply src/components/constant.js]: ok

File: src/components/contact-us.js (hunk 7 above)
Comment (joshuaobinna) [reply src/components/contact-us.js]: ok...
you're saying its not yet time for that

File: README.md (hunk 2 above)
Comment (Simpcyclassy) [reply README.md]: You actually use it to explain the summary of what's happening in the repo most times or to pass information or maybe give instructions. It does not affect the code/files in the repo but good practice for good comprehension of the repo
